components/omega/src/ocn/Tracers.h

```diff
@@ -0,0 +1,209 @@
+#ifndef OMEGA_TRACERS_H
+#define OMEGA_TRACERS_H
+//===-- ocn/Tracers.h - tracers --------------------*- C++ -*-===//
+//
+/// \file
+/// \brief Define Tracers class
+///
+/// Tracers class define tracer arrays for host and device memories and
+/// methods to handle the arrays. All of variables and methods are static
+/// because once tracers are created, they never change.
+//===----------------------------------------------------------------------===//
+
+#include "DataTypes.h"
+#include "Decomp.h"
+#include "Field.h"
+#include "Halo.h"
+
+namespace OMEGA {
+
+/// The Tracers class provides a container for tracer arrays and methods
+class Tracers {
+
+ private:
+   // total number of tracers defined at intialization
+   static I4 NumTracers;
+
+   // static storage of the tracer arrays for device and host memory spaces
+   // The arrays are 3 dimensional arrays with Tracer, Cell, and Vertical
+   // dimensions in order
+   static std::vector<Array3DReal>
+       TracerArrays; ///< TimeLevels -> [Tracer, Cell, Vert]
+   static std::vector<HostArray3DReal>
+       TracerArraysH; ///< TimeLevels -> [Tracer, Cell, Vert]
```

Comment:
In the State we (based on @mwarusz's suggestion) used:
```
Kokkos::Array<Array2DReal, MaxTimeLevels>
       LayerThickness;
```
 to handle the time levels, to avoid any device incompatibility issues with `std::vector`. Is that something we would want to do here as well?

Reply:
This was done to allow calling `LayerThickness[TimeLevel]` directly from a device kernel. However, I don't think we ended up using this capability anywhere. The approach taken in this PR is to have getter functions for tracer arrays that perform index checks and log errors. The logging alone prevents from calling them on GPU, but it is nice for safety. I think using `std::vector` here is fine and we should change the State to use it as well.